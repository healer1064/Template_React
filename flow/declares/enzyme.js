// @flow

declare module 'enzyme' {

  declare var shallow: Function

}
