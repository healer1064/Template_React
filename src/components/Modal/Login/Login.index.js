// @flow
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { withRouter } from 'react-router'
import { parse } from 'qs'
import { login } from 'src/redux/modules/auth'
import Login from './Login'

const validate = values => {

  const errors = {}
  const requiredFields = [
    'usernameInput',
    'passwordInput',
  ]
  requiredFields.forEach(field => {

    if (!values[field]) errors[field] = 'Required'

  })
  return errors

}

const onSubmit = (values, dispatch, props): Promise<*> => {

  const { usernameInput, passwordInput } = values
  const search = parse(props.location.search.substr(1))

  return login({
    username: usernameInput,
    password: passwordInput,
    redirect: search.redirect,
  }, props.history)(dispatch)

}

const reduxFormConfig = {
  form: 'loginModalForm',
  initialValues: {
    // TODO: remove these
    usernameInput: 'test@example.com',
    passwordInput: 'secret',
  },
  validate,
  onSubmit,
}


type StateProps = {
  error: string,
}

function mapStateToProps (state: GlobalReducerState): StateProps {

  const { error } = state.auth

  return {
    error,
  }

}


export default withRouter(connect(mapStateToProps)(
  reduxForm(reduxFormConfig)(Login)
))
export type ReduxProps = StateProps
