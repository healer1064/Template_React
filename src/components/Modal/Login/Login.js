// @flow
import React from 'react'
import { Link } from 'react-router'
import { Field } from 'redux-form'
import css from './Login.style.css'

import type { ReduxProps } from './Login.index'


type Props = ReduxProps & {
  handleSubmit: Function,
  location: Object,
  submitting: boolean,
  asyncValidating: boolean,
}

const Login = (props: Props): React$Element<any> => {

  const {
    handleSubmit,
    location,
    submitting,
    asyncValidating,
    error,
  } = props
  const { pathname } = location

  return (
    <div className={css.login}>
      <div className={css.header}>
        <Link
          className={css.icnCloseOffLink}
          to={{
            pathname,
            state: { modal: '' },
          }}
        >
          <button className={css.icnCloseOff} />
        </Link>
        <div className={css.title}>LOGIN</div>
      </div>
      <form
        id="loginForm"
        className={css.loginForm}
        onSubmit={handleSubmit}
      >
        <Field
          name="usernameInput"
          label="Email"
          placeholder="Email"
          component="input"
          type="email"
        />
        <Field
          name="passwordInput"
          label="Password"
          placeholder="Password"
          component="input"
          type="password"
        />
        <Link to="/?reset-password">
          <div className={css.forgotPassword}>
            Forgot Password?
          </div>
        </Link>
        <div className={css.error}>
          {error}
        </div>
        <button
          type="submit"
          disabled={submitting || asyncValidating}
        >
          {(submitting || asyncValidating) ? 'Submitting...' : 'Login'}
        </button>
      </form>
    </div>
  )


}


export default Login
