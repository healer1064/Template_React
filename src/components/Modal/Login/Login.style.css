.login {
  width: 604px;
  height: 502px;
  border-radius: 5px;
  background-color: $colorWhite;
  box-shadow: 0 6px 6px 0 rgba(0, 0, 0, 0.3);
  text-align: center;
  padding: 20px;

  .header {
    flex: 1 1 auto;
    position: relative;
    margin: 10px 0 50px;
    display: flex;
    justify-content: center;

    .icnCloseOffLink {
      position: absolute;
      left: 0;

      .icnCloseOff {
        width: 22px;
        height: 22px;
        padding: 0;
        @mixin image-url static/images/icn_close_off.png;
        border: none;
      }
    }

    .title {
      @mixin font-size 22;
      font-weight: 500;
      text-align: center;
      color: $colorDark;
    }
  }

  p {
    @mixin font-size 16;
    font-weight: 500;
  }

  .loginForm {
    display: flex;
    flex-direction: column;
    max-width: 300px;
    margin: auto;
    text-align: left;

    .forgotPassword {
      text-align: center;
      margin: 1rem 0;
    }

    button {
      background-color: $colorTheme;
      color: $colorWhite;
    }
  }
}

.error {
  color: red;
  margin: 10px auto;
}
