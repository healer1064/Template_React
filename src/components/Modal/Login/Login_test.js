// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import Login from './Login'


const mockProps = {
  handleSubmit: () => {},
  location: { pathname: '/' },
  error: '',
}


it('<Login> renders the Login content', () => {

  const wrapper = shallow(
    <Login
      {...mockProps}
      submitting={false}
      asyncValidating={false}
    />
  )

  expect(wrapper).toMatchSnapshot()

})


it('<Login> renders when form is submitting', () => {

  const wrapper = shallow(
    <Login
      {...mockProps}
      submitting
      asyncValidating={false}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
