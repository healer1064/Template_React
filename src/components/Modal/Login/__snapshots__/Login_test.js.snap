// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> renders the Login content 1`] = `
<div
  className="login"
>
  <div
    className="header"
  >
    <Link
      className="icnCloseOffLink"
      onlyActiveOnIndex={false}
      style={Object {}}
      to={
        Object {
          "pathname": "/",
          "state": Object {
            "modal": "",
          },
        }
      }
    >
      <button
        className="icnCloseOff"
      />
    </Link>
    <div
      className="title"
    >
      LOGIN
    </div>
  </div>
  <form
    className="loginForm"
    id="loginForm"
    onSubmit={[Function]}
  >
    <Field
      component="input"
      label="Email"
      name="usernameInput"
      placeholder="Email"
      type="email"
    />
    <Field
      component="input"
      label="Password"
      name="passwordInput"
      placeholder="Password"
      type="password"
    />
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/?reset-password"
    >
      <div
        className="forgotPassword"
      >
        Forgot Password?
      </div>
    </Link>
    <div
      className="error"
    />
    <button
      disabled={false}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`<Login> renders when form is submitting 1`] = `
<div
  className="login"
>
  <div
    className="header"
  >
    <Link
      className="icnCloseOffLink"
      onlyActiveOnIndex={false}
      style={Object {}}
      to={
        Object {
          "pathname": "/",
          "state": Object {
            "modal": "",
          },
        }
      }
    >
      <button
        className="icnCloseOff"
      />
    </Link>
    <div
      className="title"
    >
      LOGIN
    </div>
  </div>
  <form
    className="loginForm"
    id="loginForm"
    onSubmit={[Function]}
  >
    <Field
      component="input"
      label="Email"
      name="usernameInput"
      placeholder="Email"
      type="email"
    />
    <Field
      component="input"
      label="Password"
      name="passwordInput"
      placeholder="Password"
      type="password"
    />
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/?reset-password"
    >
      <div
        className="forgotPassword"
      >
        Forgot Password?
      </div>
    </Link>
    <div
      className="error"
    />
    <button
      disabled={true}
      type="submit"
    >
      Submitting...
    </button>
  </form>
</div>
`;
